### PR TITLE
feat: added ability to compute hash on file upload

### PR DESCRIPTION
**Pull request by sg-s**

### CODE REVIEW

File: src/data_hub/api.py [hunk 1]
```diff
@@ -265,6 +265,8 @@ def download_file(
 def upload_file(
     file_path: str,
     client=None,
+    *,
+    compute_hash: bool = False,
```
Comment (kennyjwilli) [inline src/data_hub/api.py:269]: Why defaulting off?

**Review by kennyjwilli — APPROVED** [review]

**Comment by jonrkarr** [comment]
Looks good, except for the comment about the default. Don't we want it to be on for AWS?

File: src/data_hub/api.py (hunk 1 above)
Comment (sg-s) [reply src/data_hub/api.py:269]: ok, turning on by default!

Comment (sg-s) [reply src/data_hub/api.py]: fixed

**Pull request merged**